### PR TITLE
inspec-compliance + Compliance 1.0

### PR DESCRIPTION
**Pull request by srenatus**
This change includes the necessary adaptations for having inspec's compliance plugin work with compliance 1.0.

### CODE REVIEW

**Comment by chris-rock** [comment]
We need to ensure that it also works with older versions. This extension does not work with versions pre-1.0

File: lib/bundles/inspec-compliance/api.rb [hunk 1]
```diff
@@ -9,35 +9,58 @@ module Compliance
   # API Implementation does not hold any state by itself,
   # everything will be stored in local Configuration store
   class API # rubocop:disable Metrics/ClassLength
-    # logs into the server, retrieves a token and stores it locally
-    def self.login(server, username, password, insecure)
+    # saves the api token supplied by the user
+    def self.api_token(server, refresh_token, verify, user, insecure)
```
Comment (chris-rock) [inline lib/bundles/inspec-compliance/api.rb]: we should still be able to input username and password

File: lib/bundles/inspec-compliance/README.md [hunk 2]
```diff
@@ -8,7 +8,8 @@ This extensions offers the following features:
 
 To use the CLI, this InSpec add-on adds the following commands:
 
- * `$ inspec compliance login user password` - authentication against Chef Compliance
+ * `$ inspec compliance api_token server --token TOKEN --user USER` - save the Chef Compliance API token for user
```
Comment (chris-rock) [inline lib/bundles/inspec-compliance/README.md:11]: I think this plugin should offer two methods
- handle the complete oauth flow to retrieve a token. that allows the plugin to get a refresh token as well (requires us to setup a specific client/application for inspec)
- take an api token as input

**Comment by chris-rock** [comment]
At this point of time, this PR misses a couple of things:
- usability for login should be the same, cli should take --token, --refresh-token as option
- proper fallback for legacy auth (version not determined etc)
- simple login flow

**Comment by chris-rock** [comment]
I add additions to this PRs to address this

**Comment by srenatus** [comment]
Thanks @chris-rock for the additions! :+1:

Note that when https://github.com/chef/chef-compliance/pull/767 is merged, we should probably (re)introduce another variant.

**Pull request merged**